VotingApplication/VotingApplication.Data/Model/Event.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.ComponentModel.DataAnnotations;
+
+namespace VotingApplication.Data.Model
+{
+    public enum EventType
+    {
+        Error,
+        GoToPage,
+        CreatePoll,
+        UpdateResults,
+        SetExpiry,
+        SetPollType,
+        SetQuestion,
+        SetInviteOnly,
+        SetNamedVoting,
+        SetOptionAdding,
+        AddOption,
+        UpdateOption,
+        DeleteOption,
+        AddVote,
+        DeleteVote,
+        AddBallot,
+        DeleteBallot,
+        Login,
+        Logout,
+        Register
+    }
+
+    public class Event
+    {
+        public Event(EventType eventType, Guid pollId)
+        {
+            Timestamp = DateTime.Now;
+            EventType = eventType.ToString();
+            PollId = pollId;
+        }
+
+        public long Id { get; set; }
+
+        [Required]
+        public DateTime Timestamp { get; set; }
+
+        [Required]
+        public string EventType { get; set; }
```

Comment:
Storing the enum results in an int value in the database rather than the name of the enum.

My reasoning was that  we will likely want to get data by related enums, such as "%Option" to track the lifecycle of options, or "Set%" to see how often users change the default values of the poll. This would still be possible with an additional join table, but more awkward.